### PR TITLE
[RHMWRT-221] Add OpenShift labels

### PR DESCRIPTION
**Pull request by gytis**

### CODE REVIEW

File: src/main/fabric8/route.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+metadata:
+  annotations:
+    app.kubernetes.io/vcs-uri: git@github.com:snowdrop/rest-http-example.git
+  labels:
+    app.kubernetes.io/part-of: http-api-example
+    app.kubernetes.io/runtime-version: ${spring-boot.version}
+    app.kubernetes.io/runtime: spring
```
Comment (cmoulliard) [inline src/main/fabric8/route.yml]: Should the runtime be `Spring` or `Spring Boot` instead ? 
Can you check please with Naina which value we should use ?
Is the label runtime linked to the spring or Spring boot icon to graphically display it when it is selected within the OpenShift DevConsole ?

Comment (gytis) [reply src/main/fabric8/route.yml]: @nainaz what value should we use?

Comment (gytis) [reply src/main/fabric8/route.yml]: In Rodney's table the suggested values are: nodejs | spring | thorntail | vertx

**Comment by gytis** [comment]
If we also want to add the same labels to https://github.com/snowdrop/rest-http-example/blob/sb-2.1.x/.openshiftio/application.yaml we'll need to also extend our atomist scripts to inject correct Spring Boot version.

File: src/main/fabric8/route.yml (hunk 1 above)
Comment (cmoulliard) [reply src/main/fabric8/route.yml]: FYI: The launcher catalog of the examples uses spring-boot -> https://github.com/fabric8-launcher/launcher-booster-catalog/blob/master/catalog.json#L889

Comment (gytis) [reply src/main/fabric8/route.yml]: In that case maybe we should stick to it.

Comment (gytis) [reply src/main/fabric8/route.yml]: Actually that's not correct. We should use `spring` because that's the value used to add an icon by Che

Comment (gytis) [reply src/main/fabric8/route.yml]: It's quite misleading. In here they say spring https://docs.jboss.org/display/OBST/Labels+needed+for+Examples and it is used in che here https://github.com/christianvogt/console/blob/master/frontend/public/components/catalog/catalog-item-icon.tsx#L174. However, it here it's spring-boot https://github.com/gorkem/app-labels/blob/master/labels-annotation-for-openshift.adoc

File: pom.xml [hunk 2]
```diff
@@ -266,6 +266,42 @@
...
+                      <name>app.kubernetes.io/component</name>
+                      <value>frontend</value>
+                    </property>
+                    <property>
+                      <name>app.kubernetes.io/runtime</name>
+                      <value>spring-boot</value>
+                    </property>
+                    <property>
+                      <name>app.kubernetes.io/runtime-version</name>
+                      <value>${spring-boot.version}</value>
```
Comment (gytis) [inline pom.xml]: @cmoulliard should we use Spring Boot version or our BOM version here?

**Comment by gytis** [comment]
@cmoulliard I've updated the PR. Could you give one more full review? Once we agree on how this should look like I'll update the other examples.

File: .openshiftio/application.yaml [hunk 3]
```diff
@@ -65,6 +72,14 @@ objects:
   kind: BuildConfig
   metadata:
     name: rest-http
+    labels:
+      app.kubernetes.io/part-of: http-api-example
+      app.kubernetes.io/runtime-version: "SPRING_BOOT_VERSION"
+      app.kubernetes.io/runtime: snowdrop
```
Comment (cmoulliard) [inline .openshiftio/application.yaml]: Is there a ticket opened for Openshift Devconsole to request them to display the Snowdrop icon ? @gytis

Comment (gytis) [reply .openshiftio/application.yaml]: I don't know if they show anything. We never had labels before

Comment (gytis) [reply .openshiftio/application.yaml]: I'd say we add the labels and then sort out the logos once the time comes.

Comment (nainaz) [reply .openshiftio/application.yaml]: Lets add labels and if we see that logo doesn't get displayed , we open a ticket.

**Pull request merged**